tests/tracer/test_encoders.py

```diff
@@ -348,13 +344,79 @@ def test_encoder_buffer_size_limit():
 
 
 def test_encoder_buffer_item_size_limit():
-    buffer_size = 1 << 10
-    encoder = MsgpackEncoder(buffer_size, buffer_size)
+    max_item_size = 1 << 10
+    encoder = MsgpackEncoder(max_item_size << 1, max_item_size)
 
     span = Span(tracer=None, name="test")
     trace = [span]
     encoder.put(trace)
-    trace_size = encoder.size
+    trace_size = encoder.size - 1  # This includes the global msgpack array size prefix
 
     with pytest.raises(BufferItemTooLarge):
-        encoder.put([span] * (int(buffer_size / trace_size) + 1))
+        encoder.put([span] * (int(max_item_size / trace_size) + 1))
+
+
+def test_custom_msgpack_encode_v05():
+    encoder = MsgpackEncoderV05(2 << 20, 2 << 20)
+    assert encoder.max_size == 2 << 20
+    assert encoder.max_item_size == 2 << 20
+    trace = [
+        Span(tracer=None, name="v05-test", service="foo", resource="GET"),
+        Span(tracer=None, name="v05-test", service="foo", resource="POST"),
+        Span(tracer=None, name=None, service="bar"),
+    ]
+
+    encoder.put(trace)
+    assert len(encoder) == 1
+
+    size = encoder.size
+    encoded = encoder.flush()
+    assert size == len(encoded)
+    st, ts = decode(encoded)
+
+    def filter_mut(ts):
+        return [[[s[i] for i in [0, 1, 2, 5, 7, 8, 9, 10, 11]] for s in t] for t in ts]
```

Comment:
I'm having a bit of trouble following along this list comprehension 😢 could we add a comment here explaining this?

Reply:
This is filtering out fields from the array that are non-deterministic, like trace_id, span_id, start_ns etc.. and only keeping the ones we have control over during the test runs.

Reply:
I wonder if we need to update our test agent to understand v0.5 and use that instead?

cc @Kyle-Verhoog